docs/learn/smart-contract-developers/deploy-up-with-lsp23.md

```diff
@@ -0,0 +1,447 @@
+- set the [`AddressPermissions[]`](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-6-KeyManager.md#addresspermissions) array data key
+- set the [`AddressPermissions:Permissions:<address>`](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-6-KeyManager.md#addresspermissionspermissionsaddress) data keys
+
+#### Create the LSP3 Metadata
+
+The LSP3 Metadata is a JSON object that contains information about the Universal Profile. For more information on the LSP3 Metadata, please refer to the [LSP3 Profile Metadata specification](../../standards/universal-profile/lsp3-profile-metadata.md). For the sake of simplicity, we will not create the JSON object in this guide but use random bytes instead:
+
+```typescript
+async function main() {
+  // previous code
+
+  const lsp3DataKey = ERC725YDataKeys.LSP3.LSP3Profile;
+  const lsp3DataValue = ethers.randomBytes(32); // this is just a random value for the LSP3Profile
+}
+```
+
+#### Create the Universal Receiver data key
+
+```typescript
+async function main() {
+  // previous code
+
+  const universalReceiverDataKey =
+    ERC725YDataKeys.LSP1.LSP1UniversalReceiverDelegate;
+}
+```
+
+#### Create the permissions data keys and data values
+
+Let's start by creating the permissions data keys and data values of the Universal Receiver. The Universal Receiver will have the [`SUPER_SETDATA` and `REENTRANCY`](../../standards/universal-profile/lsp6-key-manager#permissions) permissions. At the moment we are writing this guide, the combination of these two permissions is `0x0000000000000000000000000000000000000000000000000000000000060080` however this may change in the future. You can check if this is still the case by looking at the [ERC725 Inspection Tool](https://erc725-inspect.lukso.tech/key-manager).
+
+```typescript
+async function main() {
+  // previous code
+
+  const universalReceiverPermissionsKey =
+    ERC725YDataKeys.LSP6['AddressPermissions:Permissions'] +
+    UniversalReceiverAddress.slice(2);
+  const universalReceiverPermissionsValue =
+    '0x0000000000000000000000000000000000000000000000000000000000060080'; // REENTRANCY & SUPER_SETDATA permissions
+}
+```
+
+Now we will want after deployment to have a main controller that has all the permissions on the Universal Profile. We will start by adding this address under our previous constants:
+
+```typescript
+const mainController = '0x3303Ce3b8644D566271DD2Eb54292d32F1458968'; // this is just a random address
+```
+
+Then we will create the data key and data value for the permissions of the main controller:
+
+```typescript
+async function main() {
+  // previous code
+
+  const mainControllerPermissionsKey =
+    ERC725YDataKeys.LSP6['AddressPermissions:Permissions'] +
+    mainController.slice(2);
+  const mainControllerPermissionsValue = ALL_PERMISSIONS; // main controller will have all permissions on the Universal Profile
```

Comment:
Same as above, suggest writing comments above to avoid horizontal scrolling (docs page are really narrow because of the sidebars on both sides)